packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -59,7 +59,7 @@ function useInsertionPoint( {
 			let _destinationRootClientId = rootClientId;
 			let _destinationIndex;
 
-			if ( insertionIndex ) {
+			if ( insertionIndex !== undefined ) {
```

Comment:
This line fixes #32931. If `insertionIndex` is `0` then it short-circuit the logic below, which is the case when the block list is empty.